lib/puppet/pops/types/types.rb

```diff
@@ -1394,6 +1612,15 @@ def instance?(o)
         assignable?(TypeCalculator.infer(o))
       end
 
+      def iterable?
+        c = class_from_string(@runtime_type_name)
+        c.nil? ? false : c < Enumerable || c < Iterable
+      end
```

Comment:
think this makes any Ruby Enumerable be Iterable. Don't think we want that. It is too open ended.